File: src/node/magma/app/compile/value/ConstructorHeader.ts
// []
import { ConstructorHeader } from "../../../../magma/app/compile/value/ConstructorHeader";
import { FunctionHeader } from "../../../../magma/app/compile/define/FunctionHeader";
import { Platform } from "../../../../magma/app/io/Platform";
import { Definition } from "../../../../magma/app/compile/define/Definition";
import { List } from "../../../../magma/api/collect/list/List";
import { Joiner } from "../../../../magma/api/collect/Joiner";
export class ConstructorHeader implements FunctionHeader<ConstructorHeader> {
	generateWithAfterName(platform: Platform, afterName: string): string {
		return "constructor " + afterName/*string*/;
	}
	hasAnnotation(annotation: string): boolean {
		return false/*auto*/;
	}
	removeModifier(modifier: string): ConstructorHeader {
		return this/*auto*/;
	}
	addModifierLast(modifier: string): ConstructorHeader {
		return this/*auto*/;
	}
	generateWithDefinitions0(platform: Platform, definitions: string): string {
		return generateWithAfterName/*auto*/(platform/*Platform*/, "(" + definitions/*string*/ + ")");
	}
	generateWithDefinitions(platform: Platform, definitions: List<Definition>): string {
		let joinedDefinitions = definitions/*List<Definition>*/.query(/*auto*/).map((definition: Definition) => definition/*auto*/.generate(platform/*Platform*/)).collect(new Joiner(", ")).orElse("");
		return this/*auto*/.generateWithDefinitions0(platform/*Platform*/, joinedDefinitions/*auto*/);
	}
}

File: src/java/magma/app/compile/value/ConstructorHeader.java
package magma.app.compile.value;

import magma.app.compile.define.FunctionHeader;

public class ConstructorHeader implements FunctionHeader<ConstructorHeader> {
    @Override
    public String generateWithAfterName(final String afterName) {
        return "constructor " + afterName;
    }

    @Override
    public boolean hasAnnotation(final String annotation) {
        return false;
    }

    @Override
    public ConstructorHeader removeModifier(final String modifier) {
        return this;
    }

    @Override
    public ConstructorHeader addModifier(final String modifier) {
        return this;
    }
}
